File: src/code/kotlin/SameTree.kt
package code.kotlin

import code.kotlin.structures.TreeNode

class SameTree {
    fun isSameTree(p: TreeNode?, q: TreeNode?): Boolean =
        if (p == null || q == null)
            p == q
        else
            p.`val` == q.`val` && isSameTree(p.left, q.left) && isSameTree(p.right, q.right)
}

fun main() {
    val solution = SameTree()

    val p = TreeNode(1)
//    p.left = TreeNode(2)
    p.right = TreeNode(3)

    val q = TreeNode(1)
    q.left = TreeNode(2)
    q.right = TreeNode(3)

    println(solution.isSameTree(p, q))
}

File: src/code/python/same_tree.py
from src.code.python.structures.tree_node import TreeNode


class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        return (p.val == q.val and
                self.isSameTree(p.left, q.left)
                and self.isSameTree(p.right, q.right)) \
            if p is not None and q is not None \
            else p == q


if __name__ == '__main__':
    solution = Solution()
    print(solution.isSameTree(TreeNode(1, TreeNode(2), TreeNode(3)), TreeNode(1, TreeNode(2), TreeNode(3))))
